web-browsers/the-dom-and-html.md

```diff
@@ -0,0 +1,122 @@
+| DOM     | A representation of the content of a webpage in the form of a tree data structure. The root node is an object that represents the document, or the entire web page. | Document Object Model     | "The DOM shows us how content on a web page has nested relationships with each other."                                                                            |
+| Tree    | A data structure that resembles an upside down tree. Each piece of data is represented by a node, beginning with a root node.                                       | -                         | "The DOM is in a tree structure because it describes content nodes that point to other content nodes," "Tree data structures are used in some search algorithms." |
+| Node    | A unit of data in a tree, which contains the data itself and a list of other nodes it points to. Each node can only have one parent node.                           | Branch or Leaf            | "A node can point to other nodes, which make them children nodes," "All nodes have a parent node except for the root node."                                       |
+| HTML    | A markup language used to structure the content of a web page.                                                                                                      | Hypertext Markup Language |
+| Element | A unit of content in an HTML document. These are represented as nodes in the DOM.                                                                                   | HTML Element              |
+| Sibling | In a tree structure, a sibling node is a node that shares a parent with another node                                                                                |
+
+### !callout-info
+
+## Web Pages Are Documents
+
+Some contexts may refer to a web page as a document.
+
+### !end-callout
+
+## The DOM
+
+The Document Object Model, also known as **the DOM**, is a representation of the structured content on a web page.
+
+The DOM is a _tree structure_. Each part of the document (or web page) is represented as a node in the tree.
+
+### Trees
+
+Trees are data structures which organize data through nodes and branches.
+
+There is one root node. The root node has children nodes.
+
+Those children nodes can also have their own child nodes! However, each node can either have exactly zero or one parent node.
+
+![A tree structure with grey, unlabeled nodes. The root node has two children. Each node has either 0, 1, or 2 children. The tree is four layers deep.](../assets/web-browsers_the-dom-and-html_generic-tree.png)
+
+When a web page is represented in the DOM structure, the root node of the tree is always a document object that represents the document itself.
+
+Elements and pieces of content become children nodes to the document.
+
+## HTML Elements Define Document Structure
+
+We organize the content of a webpage using _HTML_.
```

Comment:
confusing mixture of italics here, but then bold in the next paragraph.

We _really_ need some clarity around what we use these to mean! 🤣 

```suggestion
We organize the content of a webpage using **HTML**.
```

Reply:
Ah! Great question and point! Very casually, I've made this rule in my head:

1. When a vocabulary word is properly defined (ideally the vocab word is subject of the sentence, and the sentence's purpose is about what it _is_ (and not what it does or contextually how it's used)), it's bold
2. When it is named otherwise, it is italic

Do you have any thoughts on that? :O

Reply:
That sounds like a reasonable distinction. I guess here it gets a little jumbled because there's the italic and the bold very near one another, and as I was puzzling over the distinction, noticed the italicized element in the follow sentence, which was presumably there solely for emphasis.

I did a quick flip through some text and technical books I had nearby, and they were fairly split on whether they used italics or bold for definitions. Personally, I'm more an italics guy, but 🤷 .

I'll try to file this away for future reference, but don't be surprised if I complain about this again sometime! 🤣